doozer/doozerlib/backend/konflux_client.py

```diff
@@ -0,0 +1,507 @@
+        rendered = template.render({
+            "source_url": https_url,
+            "revision": commit_sha,
+            "target_branch": target_branch,
+            "git_auth_secret": git_auth_secret,
+
+        })
+        obj = yaml.load(rendered)
+        # Those fields in the template are specific to an image. They need to be removed.
+        del obj["metadata"]["name"]
+        del obj["metadata"]["annotations"]["pipelinesascode.tekton.dev/on-cel-expression"]
+        # Override the generated name with the provided one
+        if generate_name:
+            obj["metadata"]["generateName"] = generate_name
+        if namespace:
+            obj["metadata"]["namespace"] = namespace
+        else:
+            del obj["metadata"]["namespace"]
+        # Set the application and component names
+        obj["metadata"]["annotations"]["build.appstudio.openshift.io/repo"] = f"{https_url}?rev={commit_sha}"
+        obj["metadata"]["labels"]["appstudio.openshift.io/application"] = application_name
+        obj["metadata"]["labels"]["appstudio.openshift.io/component"] = component_name
+
+        def _modify_param(params: List, name: str, value: Union[str, bool, list[str]]):
+            """ Modify a parameter in the params list. If the parameter does not exist, it is added.
+
+            :param params: The list of parameters.
+            :param name: The name of the parameter.
+            :param value: The value of the parameter.
+            """
+            if isinstance(value, bool):
+                value = "true" if value else "false"  # boolean value should be converted to string
+            for param in params:
+                if param["name"] == name:
+                    param["value"] = value
+                    return
+            params.append({"name": name, "value": value})
+
+        # PipelineRun parameters to override in the template
+        params = obj["spec"]["params"]
+        _modify_param(params, "output-image", output_image)
+        _modify_param(params, "skip-checks", skip_checks)
+        _modify_param(params, "image-expires-after", "6w")
+        _modify_param(params, "build-platforms", list(build_platforms))
+
+        # See https://konflux-ci.dev/docs/how-tos/configuring/customizing-the-build/#configuring-timeouts
+        obj["spec"]["timeouts"] = {"pipeline": "12h"}
+
+        # Task specific parameters to override in the template
+        for task in obj["spec"]["pipelineSpec"]["tasks"]:
+            match task["name"]:
+                case "build-images":
+                    task["timeout"] = "12h"
+                case "apply-tags":
+                    _modify_param(task["params"], "ADDITIONAL_TAGS", list(additional_tags))
+
+        # https://konflux.pages.redhat.com/docs/users/how-tos/configuring/overriding-compute-resources.html
+        # ose-installer-artifacts fails with OOM with default values, hence bumping memory limit
+        obj["spec"]["taskRunSpecs"] = [{
```

Comment:
Same as previous comment, I think storing these definitions in template files would be best. Also ok with a follow up PR